### PR TITLE
distsql: fix wrong schema version when snapshot has been set

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
As the title says.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch (release-3.1

### CODE REVIEW

File: store/mockstore/mocktikv/cop_handler_dag.go [hunk 1]
```diff
@@ -628,9 +627,6 @@ func (h *rpcHandler) fillUpData4SelectResponse(selResp *tipb.SelectResponse, dag
 	case tipb.EncodeType_TypeDefault:
 		h.encodeDefault(selResp, rows, dagReq.OutputOffsets)
 	case tipb.EncodeType_TypeChunk:
-		if dagReq.GetChunkMemoryLayout().GetEndian() != distsql.GetSystemEndian() {
```
Comment (SunRunAway) [inline store/mockstore/mocktikv/cop_handler_dag.go]: Is this check removed on purpose?

File: distsql/request_builder.go [hunk 2]
```diff
@@ -165,7 +166,11 @@ func (builder *RequestBuilder) SetFromSessionVars(sv *variable.SessionVars) *Req
 	builder.Request.NotFillCache = sv.StmtCtx.NotFillCache
 	builder.Request.Priority = builder.getKVPriority(sv)
 	builder.Request.ReplicaRead = sv.GetReplicaRead()
-	builder.Request.SchemaVar = sv.TxnCtx.SchemaVersion
+	if sv.SnapshotInfoschema != nil {
+		builder.Request.SchemaVar = sv.SnapshotInfoschema.(infoschema.InfoSchema).SchemaMetaVersion()
```
Comment (SunRunAway) [inline distsql/request_builder.go]: Could you add an integration test for it?

File: store/mockstore/mocktikv/cop_handler_dag.go (hunk 1 above)
Comment (lzmhhh123) [reply store/mockstore/mocktikv/cop_handler_dag.go]: To break the import circle. The check is not necessary.

File: distsql/request_builder.go (hunk 2 above)
Comment (lzmhhh123) [reply distsql/request_builder.go]: SchemaVar is only checked by TiFlash. None way to integration test.

Comment (SunRunAway) [reply distsql/request_builder.go]: Is it better use [infoschema.GetInfoSchema](https://github.com/pingcap/tidb/blob/9f0736e/infoschema/infoschema.go#L341) here?

File: store/mockstore/mocktikv/cop_handler_dag.go (hunk 1 above)
Comment (SunRunAway) [reply store/mockstore/mocktikv/cop_handler_dag.go]: @wshwsh12 PTAL

Comment (wshwsh12) [reply store/mockstore/mocktikv/cop_handler_dag.go]: We can remove the check.  Mocktikv is always running with TiDB in the same machine.

File: distsql/request_builder.go [hunk 3]
```diff
@@ -165,7 +166,11 @@ func (builder *RequestBuilder) SetFromSessionVars(sv *variable.SessionVars) *Req
 	builder.Request.NotFillCache = sv.StmtCtx.NotFillCache
 	builder.Request.Priority = builder.getKVPriority(sv)
 	builder.Request.ReplicaRead = sv.GetReplicaRead()
-	builder.Request.SchemaVar = sv.TxnCtx.SchemaVersion
+	if sv.SnapshotInfoschema != nil {
+		builder.Request.SchemaVar = infoschema.GetInfoSchemaBySessionVars(sv).SchemaMetaVersion()
+	} else {
+		builder.Request.SchemaVar = sv.TxnCtx.SchemaVersion
```
Comment (SunRunAway) [inline distsql/request_builder.go]: This path is redundant now.

Comment (lzmhhh123) [reply distsql/request_builder.go]: No, if we use `GetInfoSchemaBySessionVars`. It will cause some test panic. Because the info schema is not set.

Comment (SunRunAway) [reply distsql/request_builder.go]: Could you fix those tests to make code clean?

File: infoschema/infoschema.go [hunk 4]
```diff
@@ -15,6 +15,7 @@ package infoschema
 
 import (
 	"fmt"
+	"github.com/pingcap/tidb/sessionctx/variable"
```
Comment (zimulala) [inline infoschema/infoschema.go]: Please format it.

**Pull request merged**